test/extended/olm/olmv1.go

```diff
@@ -417,3 +622,96 @@ func checkFeatureCapability(oc *exutil.CLI) {
 		g.Skip("Test only runs with OperatorLifecycleManagerV1 capability")
 	}
 }
+
+// verifyAPIEndpoint runs a job to validate the given service endpoint of a ClusterCatalog
+func verifyAPIEndpoint(ctx g.SpecContext, oc *exutil.CLI, serviceURL string) {
+	jobName := fmt.Sprintf("test-catalog-endpoint-%s", rand.String(5))
+
+	tempFile, err := os.CreateTemp("", "api-test-job-*.yaml")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	defer os.Remove(tempFile.Name())
+
+	jobYAML := fmt.Sprintf(`
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  template:
+    spec:
+      containers:
+      - name: api-tester
+        image: registry.redhat.io/rhel8/httpd-24:latest
+        resources:
+          requests:
+            cpu: "10m"
+            memory: "50Mi"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -ex
+          response=$(curl -s -k "%s" || echo "ERROR: Failed to access endpoint")
```

Comment:
Follow-up is okay, but we should drop `-k` and use a proper certificate chain from service CA operator.

Reply:
The service ca is always available in a projected volume.